### PR TITLE
buildinFunc: add a strcmp function

### PR DESCRIPTION
**Pull request by zxylvlp**
Add a function needed in https://github.com/pingcap/tidb/issues/310 .

### CODE REVIEW

**Comment by coocood** [comment]
You can add the function to `expression/builtin`, so we don't need to change the parser.

**Comment by iamxy** [comment]
Don't forget to add a test first

**Comment by shenli** [comment]
Please handle the new function in typeinferer.go.
https://github.com/pingcap/tidb/blob/master/optimizer/typeinferer.go#L44

File: parser/parser.y [hunk 1]
```diff
@@ -2315,10 +2316,17 @@ FunctionCallNonKeyword:
 	{
 		$$ = &ast.FuncCallExpr{FnName: model.NewCIStr($1.(string)), Args: []ast.ExprNode{$3.(ast.ExprNode)}}
 	}
+|   "STRCMP" '(' Expression ',' Expression ')'
```
Comment (qiuyesuifeng) [inline parser/parser.y]: Alignment.

**Comment by zxylvlp** [comment]
PTAL @shenli @c4pt0r

File: expression/builtin/string_test.go [hunk 2]
```diff
@@ -167,6 +167,37 @@ func (s *testBuiltinSuite) TestLowerAndUpper(c *C) {
...
+		{[]interface{}{"", nil}, nil},
+		{[]interface{}{nil, ""}, nil},
+	}
+
+	for _, t := range tbl {
+		v, err := builtinStrcmp(t.Input, nil)
+		c.Assert(err, IsNil)
+		c.Assert(v, Equals, t.Expect)
+	}
+
```
Comment (ngaut) [inline expression/builtin/string_test.go]: Remove blank line

**Comment by shenli** [comment]
LGTM

File: expression/builtin/string.go [hunk 3]
```diff
@@ -147,6 +147,23 @@ func builtinUpper(args []interface{}, ctx map[interface{}]interface{}) (interfac
 	}
 }
 
+// See: https://dev.mysql.com/doc/refman/5.7/en/string-comparison-functions.html
+func builtinStrcmp(args []interface{}, ctx map[interface{}]interface{}) (interface{}, error) {
+	if args[0] == nil || args[1] == nil {
+		return nil, nil
```
Comment (zimulala) [inline expression/builtin/string.go]: In this case I find that MyQSL will return the error about `Unknown column 'nil' in 'field list'`. Do we need return an error ？

**Comment by zimulala** [comment]
LGTM

**Pull request merged**